src/Symfony/Component/Translation/Bridge/Crowdin/CrowdinProvider.php

```diff
@@ -0,0 +1,425 @@
+                'fileIds' => [$fileId],
+            ]),
+        ]);
+
+        if (204 === $response->getStatusCode()) {
+            throw new ProviderException(sprintf('No content in exported file %d.', $fileId), $response);
+        }
+
+        if (200 !== $response->getStatusCode()) {
+            throw new ProviderException(sprintf('Unable to export file %d translations for language "%s".', $fileId, $languageId), $response);
+        }
+
+        $export = json_decode($response->getContent(), true);
+
+        $exportResponse = $this->filesDownloader->request('GET', $export['data']['url']);
+
+        if (200 !== $exportResponse->getStatusCode()) {
+            throw new ProviderException(sprintf('Unable to download source file %d content.', $fileId), $exportResponse);
+        }
+
+        return $exportResponse->getContent();
+    }
+
+    private function downloadSourceFile(int $fileId): string
+    {
+        /**
+         * @see https://support.crowdin.com/api/v2/#operation/api.projects.files.download.get (Crowdin API)
+         * @see https://support.crowdin.com/enterprise/api/#operation/api.projects.files.download.get (Crowdin Enterprise API)
+         */
+        $response = $this->client->request('GET', sprintf('https://%s/projects/%d/files/%d/download', $this->endpoint, $this->projectId, $fileId), [
+            'headers' => [
+                'Content-Type' => 'application/json',
+            ],
+        ]);
+
+        if (200 !== $response->getStatusCode()) {
+            throw new ProviderException(sprintf('Unable to download source file %d.', $fileId), $response);
+        }
+
+        $export = json_decode($response->getContent(), true);
+
+        $exportResponse = $this->filesDownloader->request('GET', $export['data']['url']);
+
+        if (200 !== $exportResponse->getStatusCode()) {
+            throw new ProviderException(sprintf('Unable to download source file %d content.', $fileId), $exportResponse);
+        }
+
+        return $exportResponse->getContent();
+    }
+
+    private function listStrings(int $fileId, int $limit, int $offset): array
+    {
+        /**
+         * @see https://support.crowdin.com/api/v2/#operation/api.projects.strings.getMany (Crowdin API)
+         * @see https://support.crowdin.com/enterprise/api/#operation/api.projects.strings.getMany (Crowdin Enterprise API)
+         */
+        $response = $this->client->request('GET', sprintf('https://%s/projects/%d/strings', $this->endpoint, $this->projectId), [
+            'query' => [
+                'fileId' => $fileId,
+                'limit' => $limit,
+                'offset' => $offset,
+            ],
+        ]);
+
+        if (200 !== $response->getStatusCode()) {
+            $this->logger->error(sprintf('Unable to list strings for file %d in project %d. Message: "%s".', $fileId, $this->projectId, $response->getContent()));
+
+            return [];
+        }
+
+        return json_decode($response->getContent(), true)['data'];
```

Comment:
Please use `$response->toArray()` instead of `json_decode($response->getContent(), true)` (same for all similar calls)